Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2023 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import Mathlib.Analysis.ODE.PicardLindelof
+import Mathlib.Geometry.Manifold.ContMDiff
+import Mathlib.Geometry.Manifold.MFDeriv
+
+/-!
+# Integral curves of vector fields on a manifold
+
+For any continuously differentiable vector field on a manifold `M` and any chosen non-boundary point
+`x₀ : M`, an integral curve `γ : ℝ → M` exists such that `γ t₀ = x₀` and the tangent vector of `γ`
+at `t` coincides with the vector field at `γ t` for all `t` within an open interval around `t₀`.
+
+As a corollary, such an integral curve exists for any starting point `x₀` if `M` is a manifold
+without boundary.
+
+## Tags
+
+integral curve, vector field
+-/
+
+open scoped Manifold
+
+section
+
+variable
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners 𝕜 E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M] [SmoothManifoldWithCorners I M]
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E']
+  {H' : Type*} [TopologicalSpace H'] {I' : ModelWithCorners 𝕜 E' H'}
+  {M' : Type*} [TopologicalSpace M'] [ChartedSpace H' M'] [SmoothManifoldWithCorners I' M']
+
+/-- An interior point of a manifold is a point whose image in the model vector space is in the
+interior of the chart's target. -/
+def ModelWithCorners.IsInteriorPoint (x : M) := extChartAt I x x ∈ interior (extChartAt I x).target
+
+lemma ModelWithCorners.Boundaryless.isOpen_target
```

Comment:
This holds more generally for every local homeomorphism.

```/-- If I is boundaryless, every extended local homeomorphism has open target. -/
lemma LocalHomeomorph.extChart_isOpen_target [I.Boundaryless] {e : LocalHomeomorph M H} :
    IsOpen (e.extend I).target := by
  rw [extend_target, I.range_eq_univ, Set.inter_univ]
  exact I.continuous_symm.isOpen_preimage _ e.open_target
  ```
(My suggested name is probably not ideal.) #find_home suggests this should go into SmoothManifoldWithCorners.lean.

Reply:
Included your lemma. Thanks! I'm keeping it all in one file for now but will move lemmas to other files at the end where appropriate.